app/models/spree/invoice.rb

```diff
@@ -0,0 +1,128 @@
+class Spree::Invoice < ActiveRecord::Base
+  belongs_to :invoiceable, polymorphic: true
+  validates :invoiceable, presence: true
+
+  before_save :set_date
+  before_save :set_number, if: :use_sequential_number?
+  after_save :increase_invoice_number, if: :use_sequential_number?
```

Comment:
Ah, I got it. You don't want to have skipped invoice numbers, if something went wrong. Check!